### PR TITLE
tools: fixed errors

### PR DESCRIPTION
**Pull request by mdanilov**
Fixed:
1. persistence_db_viewer wrong handle resource list
2. persistence_client_tool use not terminated strings

### CODE REVIEW

**Comment by gunnarx** [comment]
Interesting. I think even a few more places need fixing also - see comments

If you update, may I suggest:
tool: ...
Instead of
tools: ...
Since the program is a single tool?

File: tools/persistence_client_tool.c [hunk 1]
```diff
@@ -291,7 +291,7 @@ int main(int argc, char *argv[])
 		   case 'a':   // application name
 		   {
 		      size_t len = strlen(optarg);
-            appName = malloc(len);
+            appName = malloc(len + 1);
             if(appName != NULL)
             {
                memset(appName, 0, len);
```
Comment (gunnarx) [inline tools/persistence_client_tool.c]: Here should be len + 1 now? Like you did on line 380.

File: tools/persistence_client_tool.c [hunk 2]
```diff
@@ -302,7 +302,7 @@ int main(int argc, char *argv[])
 		   case 'r':   // resource ID
          {
             size_t len = strlen(optarg);
-            resourceID = malloc(len);
+            resourceID = malloc(len + 1);
             if(resourceID != NULL)
             {
                memset(resourceID, 0, len);
```
Comment (gunnarx) [inline tools/persistence_client_tool.c]: Here

File: tools/persistence_client_tool.c [hunk 3]
```diff
@@ -313,7 +313,7 @@ int main(int argc, char *argv[])
          case 'p':   // payload to write
          {
             size_t len = strlen(optarg);
-            payloadBuffer = malloc(len);
+            payloadBuffer = malloc(len + 1);
             if(payloadBuffer != NULL)
             {
                memset(payloadBuffer, 0, len);
```
Comment (gunnarx) [inline tools/persistence_client_tool.c]: Here

File: tools/persistence_client_tool.c [hunk 4]
```diff
@@ -324,7 +324,7 @@ int main(int argc, char *argv[])
          case 'f':   // filename to read data from, write data to
          {
             size_t len = strlen(optarg);
-            fileName = malloc(len);
+            fileName = malloc(len + 1);
             if(fileName != NULL)
             {
                memset(fileName, 0, len);
```
Comment (gunnarx) [inline tools/persistence_client_tool.c]: Here

**Review by gunnarx — CHANGES_REQUESTED** [review]
As noted above.

**Comment by gunnarx** [comment]
db_viewer change looks alright to me (resource list handling a bit hard to follow but if assuming it was correct before the change)

**Comment by mdanilov** [comment]
> db_viewer change looks alright to me (resource list handling a bit hard to follow but if assuming it was correct before the change)

Actually I found the mistake in resource list handling. Fixed in my next commit.
